src/scope/models/version.ts

```diff
@@ -52,6 +53,42 @@ type ExternalHead = { head: Ref; laneId: LaneId };
 type SquashData = { previousParents: Ref[]; laneId: LaneId };
 type VersionOrigin = { id: { scope: string; name: string }; lane?: { scope: string; name: string; hash: string } };
 
+export type DependencyNode = {
+  pkgId: string;
+  attr: {
+    component?: {
+      scope: string;
+      name: string;
+    };
+  } & PackageInfo;
+};
+
+export type DependencyEdge = {
+  id: string;
+  neighbours: DependencyNeighbour[];
+  attr: {
+    pkgId: string;
+  };
+};
+
+export type DependencyNeighbour = {
+  id: string;
+  type: 'prod' | 'optional';
+};
+
+export type DirectDependency = {
+  name: string;
+  specifier: string;
+  nodeId: string;
+};
+
+export type DependenciesGraph = {
```

Comment:
I would consider making this a class and not just a type, so you can make an API for it.

Reply:
Also maybe make it an instance of clear graph? or a composition of cleargraph (so you will put the node and edges inside on a cleargraph instance, then you can make APIs that calls cleargraph internally)

Reply:
Converted to a class. Conversion to cleargraph will be done in a separate PR.